src/asm3/onlineform.py

```diff
@@ -374,9 +374,22 @@ def get_onlineform_html(dbo: Database, formid: int, completedocument: bool = Tru
     if completedocument:
         h.append(asm3.utils.nulltostr(form.FOOTER))
         footer = get_onlineform_footer(dbo)
-        h.append(footer.replace("$$TITLE$$", form.NAME))
+        extrajs = '\n'.join(['<script>',
+            'let ro = new ResizeObserver(function(e) {',
+            '   window.parent.postMessage(document.querySelector("html").offsetHeight, "*");',
+            '});',
+            'ro.observe(document.querySelector("html"));',
+            '</script>'
+        ])
+        h.append(footer.replace("$$TITLE$$", form.NAME).replace("</body>", extrajs + "</body>"))
     return "\n".join(h)
 
+def get_onlineform_js(dbo: Database, formid: int) -> str:
+    """ Returns js that outputs online form into a host div """
+    js = asm3.utils.read_text_file("%s/static/js/onlineform_embed.js" % dbo.installpath)
+    js = js.replace('{SERVICE_URL}', SERVICE_URL).replace('{TOKEN_FORMID}', str(formid))
```

Comment:
tiny readability thing but I'd do the replaces on separate lines - and use double quotes